Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -521,6 +521,12 @@ def self.livecheck_url_to_string(livecheck_url, package_or_resource)
       when :homepage
         package_or_resource.homepage unless package_or_resource.is_a?(Resource)
       end
+
+      if livecheck_url.is_a?(Symbol) && !livecheck_url_string
+        raise ArgumentError, "`url #{livecheck_url.inspect}` does not reference a checkable URL"
+      end
+
+      livecheck_url_string
```

Comment:
```suggestion
      else 
        raise ArgumentError, "`url #{livecheck_url.inspect}` does not reference a checkable URL"
      end
```

does this work or something else where it can be part of the case statement? If not: feels like this should probably also `raise` on invalid Symbols?

Reply:
I initially tried to implement it as part of the case statement (e.g., `when Symbol` or `else`) but that branch wouldn't be reached if `livecheck_url` is one of the specified symbols and the result is `nil`. For example, this occurs if you use `url :url` in a formula `livecheck` block (instead of `url :stable`).

We could use `else` to account for invalid symbols but [the `#url` method already does this](https://github.com/Homebrew/brew/blob/ddfe04a6013131d4f70a278360f4991011dca679/Library/Homebrew/livecheck.rb#L174), so that branch may never get exercised in practice. However, the general approach for the `livecheck` block DSL methods is to accept any value (of the expected type) and have livecheck validate the value when it's used. We could modify `#url` to accept any symbol and do the validation in `#livecheck_url_to_string` instead.

If that makes sense to you, I'll add it here.

Reply:
@samford Thanks for context, fine as-is then!